registryoverwrite.go

```diff
@@ -0,0 +1,181 @@
+	copy(sig[:], sigBytes)
+
+	// Verify the signature on the response.
+	srv := modules.SignedRegistryValue{
+		RegistryValue: modules.RegistryValue{
+			Tweak:    dataKey,
+			Data:     data,
+			Revision: rhg.Revision,
+			Type:     rhg.Type,
+		},
+		Signature: sig,
+	}
+	err = srv.Verify(spk.ToPublicKey())
+	if err != nil {
+		return nil, 0, errors.AddContext(err, "signature returned by portal does not match")
+	}
+
+	// Return the data and revision number.
+	return data, rhg.Revision, nil
+}
+
+// OverwriteRegistry will overwrite an existing registry entry, replacing it
+// with the provided value.
+//
+// WARNING: Improper use of this function can cause data loss, and has caused
+// users data loss in the past. The common mistake is to first read from the
+// registry entry, see that nothing is there, and then call OverwriteRegistry
+// with new data. If the initial read fasely returned 404 or was subject to a
+// race condition which prevented it from seeing the most recent data (both
+// very common on Skynet), the resulting call to OverwriteRegistry can cause
+// data loss. You should ONLY use OverwriteRegistry on registry entries that
+// are not storing user data.
+//
+// An example of an acceptable use of this function is to deploy application
+// updates to skynet. Because a newly updated application is fully independent
+// of previously deployed applications, it is okay if previous data gets lost.
+//
+// An example of an unacceptable use of this function would be to update a list
+// of a user's files. When updating the list, the new update depends on
+// information previously stored in the registry (the new update is a
+// combination of the previous data + the new data). Using OverwriteRegistry
+// for this purpose WILL cause data loss. Instead, use a pattern where the
+// registry revision number is set on the read, guaranteeing that the
+// subsequent write will not destroy data in the event that the read was
+// subject to a network error or network race condition.
+//
+// WARNING: This function does set any primary hosts, and therefore is not
+// suitable for registry entries that people may wish to subscribe to. If
+// people do attempt to subscribe to this registry entry, they may miss updates
+// or receive updates at a significant delay.
+//
+// NOTE: Because this function is uploading signed data, we do not need
+// substantial evidence of honesty from the portal. One thing that we could do
+// to enforce a bit more accountability on the portal is require the portal to
+// provide some signatures from hosts which demonstrate that the hosts received
+// the update. That eliminates deceptions where the portal pretends to execute
+// the registry update but actually does not. Because the data is signed, there
+// is no opportunity for the portal to place false or bad data on the network.
+func (sc *SkynetClient) OverwriteRegistry(secretKey crypto.SecretKey, dataKey crypto.Hash, data []byte) error {
```

Comment:
maybe it makes sense to call this `OverwriteRegistryUnsafe`, I think the warnings are great but at the same time I think it's unfortunate they "only make it to the docstring", so if you fail to check the docstring you might be completely oblivious to the hidden dangers with using this method on the client